ci.py

```diff
@@ -178,12 +206,158 @@ def kill(signal, frame):
+            else:
+                services_to_update.append(service_name)
+            for service in services_to_update:
+                image = compose_services[service]['image']
+                image = image.split(':')[0]
+                image += ":ci-cd"
+                compose_services[service]['image'] = image
+    # Update compose version
+    compose_dict['version'] = '2'
+    try:
+        with open('docker-compose.yml', 'w') as docker_compose:
+            yaml.dump(compose_dict, docker_compose, default_flow_style=False)
+    except:
+        raise FileWriteException('Error writing modified dictionary to docker-compose.yml')
+
+
 def handle_pull_request(files, modules, tags):
     if modules:
         run_build(modules)
     else:
         print('No modules to build.')
+    update_docker_compose(modules)
+    run_docker_compose()
+    wait_for_init_jobs()
+    run_smoke_tests()
+
+
+def get_current_init_status(docker_id):
+    init_status = ['docker', 'inspect', '-f', '{{ .State.ExitCode }}:{{ .State.Status }}', docker_id]
+    p = subprocess.Popen(init_status, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+
+    def kill(signal, frame):
+        p.kill()
+        print()
+        print('killed!')
+        sys.exit(1)
+
+    signal.signal(signal.SIGINT, kill)
+
+    output, err = p.communicate()
+
+    if p.wait() != 0:
+        print('getting current status failed')
+        return False
+    status_output = output.rstrip()
+
+    exit_code, status = status_output.split(":", 1)
+    if exit_code == "0" and status == "exited":
+        return True
+    return False
+
+
+def output_docker_logs():
+    docker_logs = ['docker-compose', 'logs']
+
+    docker_logs_process = subprocess.Popen(docker_logs, stdin=subprocess.PIPE)
+
+    def kill(signal, frame):
+        docker_logs_process.kill()
+        print()
+        print('killed!')
+        sys.exit(1)
+
+    signal.signal(signal.SIGINT, kill)
+    if docker_logs_process.wait() != 0:
+        print('Error listing logs')
+
+
+def output_docker_ps():
+    docker_ps = ['docker', 'ps', '-a']
+
+    docker_ps_process = subprocess.Popen(docker_ps, stdin=subprocess.PIPE)
+
+    def kill(signal, frame):
+        docker_ps_process.kill()
+        print()
+        print('killed!')
+        sys.exit(1)
 
+    signal.signal(signal.SIGINT, kill)
+    if docker_ps_process.wait() != 0:
+        print('Error running docker ps')
+
+
+def get_docker_id(init_job):
+    docker_id = ['docker-compose', 'ps', '-q', init_job]
+
+    p = subprocess.Popen(docker_id, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+
+    def kill(signal, frame):
+        p.kill()
+        print()
+        print('killed!')
+        sys.exit(1)
+
+    signal.signal(signal.SIGINT, kill)
+
+    output, err = p.communicate()
+
+    if p.wait() != 0:
+        print('error getting docker id')
+        return ""
+    return output.rstrip()
+
+
+def wait_for_init_jobs():
+    init_status_dict = {"mysql-init": False,
```

Comment:
i could do that as well as the alarm init. I was thinking of adding a check later on to check that all containers are in a healthy state or exited with 0 so that would catch the other inits and containers